### PR TITLE
fix: Modal width out of screen

### LINKED ISSUE #25554: Modal jumps out of screen
- [ x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/modal-out-of-screen-6vv7l)

### Steps to reproduce
1.Create modal,
2.Add props: centered and width=1000 (should be more than 768)
3.Open modal and change browser width from > 1000 to <1000 and then <768

### What is expected?
Modal should remain in the middle of the screen

### What is actually happening?
when screen width becomes <1000 (defined modal width) it jumps down out of screen and scroll bar appears

| Environment | Info |
|---|---|
| antd | 4.4.1 |
| React | 16.12 |
| System | windows |
| Browser | chrome 83 |

---
- it happens also in version 3.*
- possible solution:
```less
.ant-modal-centered {
  white-space: nowrap; // prevent wrapping modal after :before element, that has screen height
  > .ant-modal {
    white-space: normal; // restore wrapping for internal elements
  }
}
```

### PR DESCRIPTION
**Pull request by tanmoyopenroot**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

closes #25554 

### 💡 Background and solution
When then `width` of the `browser` is less than `1000` and when `width` prop is added with a value of `1000`, a scrollbar appears.

Added `max-width` to the modal to fix the problem

Screenshot 2020-07-23 at 7 04 11 AM

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Fix Modal jump out of screen in some situations. |
| 🇨🇳 Chinese |  修复 Modal 在某些情况下溢出视窗的问题。   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is not needed
- [x] Demo is provided
- [x] TypeScript definition is not needed
- [x] Changelog is provided

-----
[View rendered components/modal/demo/width.md](https://github.com/tanmoyopenroot/ant-design/blob/25554-fix-modal-width/components/modal/demo/width.md)

### CODE REVIEW

**Comment by 07akioni** [comment]
Can `overlayStyle={{ maxWidth: 1000 }}` work?

**Comment by tanmoyopenroot** [comment]
> Can `overlayStyle={{ maxWidth: 1000 }}` work?

I don't think it will work with `maxWidth: 1000` should be in terms of `percentage` like `max-width: 100%`, so that it does not overflow when `modal width` is more than the `browser`.

File: components/modal/style/modal.less [hunk 1]
```diff
@@ -96,7 +98,8 @@
     padding: @modal-footer-padding-vertical @modal-footer-padding-horizontal;
     text-align: right;
     background: @modal-footer-bg;
-    border-top: @modal-footer-border-width @modal-footer-border-style @modal-footer-border-color-split;
+    border-top: @modal-footer-border-width @modal-footer-border-style
```
Comment (07akioni) [inline components/modal/style/modal.less:101]: keep original style

File: components/modal/style/modal.less [hunk 2]
```diff
@@ -81,7 +82,8 @@
     padding: @modal-header-padding;
     color: @text-color;
     background: @modal-header-bg;
-    border-bottom: @modal-header-border-width @modal-header-border-style @modal-header-border-color-split;
+    border-bottom: @modal-header-border-width @modal-header-border-style
```
Comment (07akioni) [inline components/modal/style/modal.less:85]: keep original style

**Review by 07akioni — COMMENTED** [review]
the `white-space` is still the problem. if i want to keep the modal width and write
`<Modal overlayStyle={{ maxWidth: 'unset' }} width={998} />`. it will still jumps out of screen.

File: components/modal/style/modal.less (hunk 1 above)
Comment (afc163) [reply components/modal/style/modal.less:101]: It should be formatted by prettier.

**Comment by tanmoyopenroot** [comment]
> `<Modal overlayStyle={{ maxWidth: 'unset' }} width={998} />`. it will still jumps out of screen.

Well, if we `unset` the `maxWidth`, then it's gonna take the value from the `width` prop.
Thereby causing an overflow.

**Comment by 07akioni** [comment]
> > `<Modal overlayStyle={{ maxWidth: 'unset' }} width={998} />`. it will still jumps out of screen.
> 
> Well, if we `unset` the `maxWidth`, then it's gonna take the value from the `width` prop.
> Thereby causing an overflow when the `browser` width is less than `998`.

I mean in this case, the current implementation still not enough.

You should set container's `white-space` to `nowrap` and body's `white-space` to `unset`. Checkout the original issue and take a look at his solution.

These fixes can be integrated.

**Pull request merged**